ultrasonic/src/main/kotlin/org/moire/ultrasonic/service/MediaPlayerService.kt

```diff
@@ -136,6 +148,99 @@ class MediaPlayerService : Service() {
         }
     }
 
+    override fun onGetRoot(
+        clientPackageName: String,
+        clientUid: Int,
+        rootHints: Bundle?
+    ): MediaBrowserServiceCompat.BrowserRoot {
+
+        // Returns a root ID that clients can use with onLoadChildren() to retrieve
+        // the content hierarchy. Note that this root isn't actually displayed.
+        return MediaBrowserServiceCompat.BrowserRoot(MEDIA_BROWSER_ROOT_ID, null)
+    }
+
+    override fun onLoadChildren(
+        parentMediaId: String,
+        result: MediaBrowserServiceCompat.Result<List<MediaBrowserCompat.MediaItem>>
+    ) {
+
+        val mediaItems: MutableList<MediaBrowserCompat.MediaItem> = mutableListOf()
+
+        if (MEDIA_BROWSER_ROOT_ID == parentMediaId) {
+            // Build the MediaItem objects for the top level,
+            // and put them in the mediaItems list...
+
+            var albumList: MediaDescriptionCompat.Builder = MediaDescriptionCompat.Builder()
+            albumList.setTitle("Browse Albums").setMediaId(MEDIA_BROWSER_ALBUM_LIST_ROOT)
+            mediaItems.add(
+                MediaBrowserCompat.MediaItem(
+                    albumList.build(),
+                    MediaBrowserCompat.MediaItem.FLAG_BROWSABLE
+                )
+            )
+        } else if (MEDIA_BROWSER_ALBUM_LIST_ROOT == parentMediaId) {
+            executorService.execute {
+                val musicService = getMusicService()
+
+                val musicDirectory: MusicDirectory = musicService.getAlbumList2(
+                    "alphabeticalByName", 10, 0, null
+                )
```

Comment:
This part happens when someone clicks on "Browse Albums" from the main menu. As you can see here I just download 10 albums using one call I know works from the `musicService`. Usually there are checks for ID3 settings and other things. This should be more robust. 

Also, do we even want to download? Maybe Android Auto should only use Offline Mode? Pretty much guaranteed users won't be using their home WiFi here. Also, in a car we don't really want to show hundreds of albums in a user's collection.

This is another case where we need some community agreement

Reply:
I think we must support online mode. I always use mobile data to play music in my car (that's why I have a subsonic server in the first place, so I can access my collection on the go). I think I'm not alone with this, also unlimited data plans are becoming more common...
I agree showing many-many albums at a time is pointless, I can't imagine browsing for minutes while driving. But the search (also voice search) becomes more important...

Reply:
Okay cool, this is valuable feedback. I was burned by the endless download bug, so I tend to download all my music while at home and then only ever run Ultrasonic in "Offline Mode" when I leave the house. I wasn't sure what was common. Okay, I'll pursue ways to make it usable assuming hundreds of albums accessible online.